backend/src/middleware/auth.ts

```diff
@@ -13,28 +15,31 @@ export const jwtCheck = auth({
 });
 
 export const jwtParse = async (req: Request, res: Response, next: NextFunction) => {
-  const { authorization } = req.headers;
-  if (!authorization || !authorization.startsWith('Bearer ')) {
-    return res.sendStatus(401);
-  }
-  const token = authorization.split(' ')[1];
-
   try {
+    const { authorization } = req.headers;
+    if (!authorization || !authorization.startsWith('Bearer ')) {
+      throw new AppError('No token provided', 401);
+    }
+    const token = authorization.split(' ')[1];
+
     const decoded = jwt.decode(token);
     if (!decoded) {
-      return res.sendStatus(401);
+      throw new AppError('Invalid token', 401);
     }
     const auth0Id = decoded.sub;
     const user = await User.findOne({ auth0Id });
     if (!user) {
-      return res.sendStatus(401);
+      throw new AppError('User not found', 401);
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider more specific error message</b></div>
<div id="fix">

Consider using a more specific error message to provide clearer feedback. Perhaps include the `auth0Id` in the error message to aid in debugging.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -32 +32 @@
 -      throw new AppError('User not found', 401);
 +      throw new AppError(`User not found for auth0Id: ${auth0Id}`, 401);
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/Food-Ordering-App/pull/47#issuecomment-2439370000>#eef1bf</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged